cmd/gossamer/main.go

```diff
@@ -61,15 +67,65 @@ func main() {
 	}
 }
 
+func LvlFromInt(lvlUint int) (log.Lvl, error) {
+	switch lvlUint {
+	case 5:
+		return log.LvlTrace, nil
+	case 4:
+		return log.LvlDebug, nil
+	case 3:
+		return log.LvlInfo, nil
+	case 2:
+		return log.LvlWarn, nil
+	case 1:
+		return log.LvlError, nil
+	case 0:
+		return log.LvlCrit, nil
+	default:
+		return log.LvlDebug, fmt.Errorf("Unknown level: %v", lvlUint)
+	}
+}
+
+func StartLogger(ctx *cli.Context) error {
+	logger := log.Root()
+	handler := logger.GetHandler()
+
+	level, err := strconv.Atoi(ctx.String("verbosity"))
+	if err != nil {
+		lvl, err := log.LvlFromString(ctx.String("verbosity"))
+		if err != nil {
+			return err
+		}
+
+		log.Root().SetHandler(log.LvlFilterHandler(lvl, handler))
+
+	} else {
+		lvl, err := LvlFromInt(level)
+		if err != nil {
+			return err
+		}
+
+		log.Root().SetHandler(log.LvlFilterHandler(lvl, handler))
+	}
+
+	return nil
+}
+
 // gossamer is the main entrypoint into the gossamer system
 func gossamer(ctx *cli.Context) error {
-	srvlog := log.New(log.Ctx{"blockchain": "gossamer"})
+
+	err := StartLogger(ctx)
+	if err != nil {
+		log.Error("verbosity level error", "err", err)
```

Comment:
Rather return the error and it will just be spit out on the command line. We probably shouldn't proceed if the logger fails.